### PR TITLE
Add docs page for rancher-webhook

### PR DESCRIPTION
**Pull request by KevinJoiner**
Fixes https://github.com/rancher/rancher-docs/issues/580

### CODE REVIEW

**Review by MbolotSuse — CHANGES_REQUESTED** [review]

File: docs/reference-guides/rancher-webhook.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+title: Rancher Webhook
+---
+
+Rancher-Webhook is an essential component of Rancher that works in conjunction with Kubernetes to enhance security and enable various features for Rancher managed clusters. 
+This documentation provides an overview of the webhook, explains its purpose, and highlights its limitations.
+
+## What does it do?
+
+The Rancher-Webhook integrates with Kubernetes' extensible admission controllers, as described in the [Kubernetes documentation](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/). This integration allows Rancher-Webhook to inspect specific request sent to the Kubernetes API server and add custom validation and mutations to the request that are specific to Rancher. Users can view resources validated by Rancher-Webhook with `kubectl get validatingwebhookconfigurations rancher.cattle.io`.
```
[the remark this answers is not in the document]
Comment (MbolotSuse) [reply docs/reference-guides/rancher-webhook.md:10]: We should also consider not telling people to look at these resources at all, and instead telling them to look for the version of the deployed chart and look at the docs instead.

[the remark this answers is not in the document]
Comment (KevinJoiner) [reply docs/reference-guides/rancher-webhook.md:10]: I will update this comment to only convey to the user that rancher-werbhook owns the rancher.cattle.io resource

File: docs/reference-guides/rancher-webhook.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+This documentation provides an overview of the webhook, explains its purpose, and highlights its limitations.
+
+## What does it do?
+
+The Rancher-Webhook integrates with Kubernetes' extensible admission controllers, as described in the [Kubernetes documentation](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/). This integration allows Rancher-Webhook to inspect specific requests sent to the Kubernetes API server and add custom validation and mutations to the requests that are specific to Rancher. Rancher-Webhook manages the resources to be validated using the `rancher.cattle.io` `ValidatingWebhookConfiguration` and the `rancher.cattle.io` `MutatingWebhookConfiguration` and will override any manual edits.
+Rancher deploys Rancher-Webhook as a separate deployment and service in both local and downstream clusters. Rancher manages Rancher-Webhook using Helm, and it's important to note that Rancher may override modifications made by users to the Helm release.
+
+## Why do we need it?
+
+Rancher-Webhook is crucial for Rancher to protect clusters against malicious attacks and enable various features.
```
Comment (martyav) [inline docs/reference-guides/rancher-webhook.md:15]: Maybe change "various" to something that emphasizes their importance, like "critical features", "key features", etc

File: docs/reference-guides/rancher-webhook.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+Rancher relies on the Rancher-Webhook as an integral part of its functionality. Without the webhook, Rancher would not be a complete product. 
+It provides essential protection for Rancher-managed clusters, preventing security vulnerabilities and ensuring the consistency and stability of the cluster.
+
+## Common Issues
+
+### EKS Cluster with Calico CNI
+
+Users running an EKS cluster with Calico CNI may run into errors when the Kubernetes API server attempts to contact the Rancher-Webhook. 
+One workaround for this issue [documented by calico](https://docs.tigera.io/calico/latest/getting-started/kubernetes/managed-public-cloud/eks#install-eks-with-calico-networking) involves setting `hostNetwork=true` for the webhook deployment. Users can change this using the Helm commands below on the affected clusters.
```
Comment (btat) [inline docs/reference-guides/rancher-webhook.md:21]:
```suggestion
One workaround for this issue [documented by Calico](https://docs.tigera.io/calico/latest/getting-started/kubernetes/managed-public-cloud/eks#install-eks-with-calico-networking) involves setting `hostNetwork=true` for the webhook deployment. Users can change this using the Helm commands below on the affected clusters.
```

**Comment by btat** [comment]
@martyav @LucasSaintarbor please update the corresponding files in versioned_docs/version-2.7 as well before merging.

**Review by martyav — APPROVED** [review]

**Comment by martyav** [comment]
@KevinJoiner I just noticed this PR is pointing to `main`. Should this point to the release branch -- or should we keep it on `main` since the webhook is already out? If so, we should probably remove the next-release tag to make it clear that this doesn't depend on v2.7.5.

**Comment by KevinJoiner** [comment]
@martyav This PR targets the main branch because it was the default branch. I don't know what the branching strategy is for the repo. But this documentation applies to all Rancher versions v2.7.0 and greater and does not depend specifically on v2.7.5

**Comment by martyav** [comment]
@KevinJoiner since it doesn't depend on 2.7.5, we'll keep it on `main`

**Comment by btat** [comment]
@MbolotSuse Can you please review this again, it's blocking the PR from being merged.

**Comment by MbolotSuse** [comment]
@btat I dismissed my stale review, you can now merge if you would like.

**Comment by btat** [comment]
> @btat I dismissed my stale review, you can now merge if you would like.

Thank you!

**Pull request merged**